public/app/features/provisioning/HomePage.tsx

```diff
@@ -0,0 +1,169 @@
+import { MigrateToRepository } from './MigrateToRepository';
+import { RepositoryActions } from './RepositoryActions';
+import { RepositoryOverview } from './RepositoryOverview';
+import { RepositoryResources } from './RepositoryResources';
+import { useDeletecollectionRepositoryMutation, useGetFrontendSettingsQuery } from './api';
+import { useRepositoryList } from './hooks';
+import { checkSyncSettings } from './utils';
+
+const appEvents = getAppEvents();
+
+enum TabSelection {
+  Overview = 'overview',
+  Resources = 'resources',
+  Files = 'files',
+  GettingStarted = 'getting-started',
+  Repositories = 'repositories',
+}
+
+const connectedTabInfo = [
+  { value: TabSelection.Overview, label: 'Overview', title: 'Repository overview' },
+  { value: TabSelection.Resources, label: 'Resources', title: 'Resources saved in grafana database' },
+  { value: TabSelection.Files, label: 'Files', title: 'The raw file list from the repository' },
+  { value: TabSelection.GettingStarted, label: 'Getting started', title: 'Getting started' },
+];
+
+const disconnectedTabInfo = [
+  { value: TabSelection.Repositories, label: 'Repositories', title: 'List of repositories' },
+  { value: TabSelection.GettingStarted, label: 'Getting started', title: 'Getting started' },
+];
+
+export default function HomePage() {
+  const [items, isLoading] = useRepositoryList({ watch: true });
+  const settings = useGetFrontendSettingsQuery();
+  const [deleteAll, deleteAllResult] = useDeletecollectionRepositoryMutation();
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [showExportModal, setShowExportModal] = useState(false);
+  const [showMigrateModal, setShowMigrateModal] = useState(false);
+  const [instanceConnected] = checkSyncSettings(settings.data);
+  const [activeTab, setActiveTab] = useState<TabSelection>(
+    instanceConnected ? TabSelection.Overview : TabSelection.Repositories
+  );
+
+  useEffect(() => {
+    if (deleteAllResult.isSuccess) {
+      appEvents.publish({
+        type: AppEvents.alertSuccess.name,
+        payload: ['All configured repositories deleted'],
+      });
+    }
+  }, [deleteAllResult.isSuccess]);
+
+  useEffect(() => {
+    setActiveTab(instanceConnected ? TabSelection.Overview : TabSelection.Repositories);
+  }, [instanceConnected]);
+
+  // Early return for onboarding
+  if (!items?.length && !isLoading) {
+    return <GettingStartedPage />;
+  }
+
+  const onConfirmDelete = () => {
+    deleteAll({});
+    setShowDeleteModal(false);
+  };
+
+  const renderTabContent = () => {
+    if (!instanceConnected) {
+      switch (activeTab) {
+        case TabSelection.Repositories:
+          return <FolderRepositoryList items={items ?? []} />;
+        case TabSelection.GettingStarted:
+          return <GettingStarted />;
+        default:
+          return null;
+      }
+    }
+
+    // At this point we know items exists and has length > 0
+    const repo = items![0];
```

Comment:
```suggestion
    const repo = items?.[0];
    if (!repo) {
      return null;
    }
```

Repo can by undefined according to the types, so we need to handle that.